common/src/main/java/com/wynntils/handlers/item/ItemHandler.java

```diff
@@ -248,6 +251,27 @@ private ItemAnnotation calculateAnnotation(ItemStack itemStack, StyledText name)
         return annotation;
     }
 
+    private StyledText simplifyName(StyledText name) {
+        String str = name.getString();
+
+        final Pattern[] PATTERNS = {
+            Pattern.compile("^§6(?:Buying|Selling) [^ ]+ (.+)§6 for .+ Each$"),
+            Pattern.compile("^§6(?:Buying|Selling) [^ ]+ (.+)$"),
+            Pattern.compile("^§7§l[^ ]+x (.+)$")
+        };
+
+        for (Pattern p : PATTERNS) {
+            Matcher m = p.matcher(str);
```

Comment:
FYI you can call `matcher` directly from `StyledText`, although in this example it'd be more expensive because `getString` isn't cached.

Reply:
I did it this way because there was less code that could break. I might as well leave it because of the slight optimization like you mentioned and so all the logic would fit on screen.